pkg/controllers/subnetset/subnetset_controller.go

```diff
@@ -75,10 +75,10 @@ func (r *SubnetSetReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 					return ResultRequeue, err
 				}
 				if obj.Spec.AccessMode == "" {
-					obj.Spec.AccessMode = v1alpha1.AccessMode(vpcNetworkConfig.DefaultSubnetAccessMode)
+					obj.Spec.AccessMode = v1alpha1.AccessMode(v1alpha1.AccessModePrivate)
```

Comment:
I think we should check if this Accessmode is set for POD or VM
For POD subnetset, we should use vpcNetworkconfiguration.PodSubnetAccessMode
like https://github.com/vmware-tanzu/nsx-operator/pull/627/files#diff-ea8fdc4a9adcc30f1ee646f03f10d565526b3a2b6fec57ae9abaa316c14b8d89R114
we check VM subnetset.

Reply:
updated.

Reply:
I mean you need to check this CR is pod-default subnetset or vm-default subnetset.  we should not set  pod-default subnetset using vpcNetworkconfiguration.PodSubnetAccessMode

Reply:
vpcNetworkconfiguration.PodSubnetAccessMode is only for pod-default subnetset

Reply:
vm-default subnetset can be set private if user doesn't set it

Reply:
Thanks. Updated.